xds/src/main/java/io/grpc/xds/ClientXdsClient.java

```diff
@@ -164,10 +164,12 @@ final class ClientXdsClient extends XdsClient implements XdsResponseHandler, Res
           : Boolean.parseBoolean(System.getProperty("io.grpc.xds.experimentalEnableLeastRequest"));
   @VisibleForTesting
   static boolean enableCustomLbConfig =
-      !Strings.isNullOrEmpty(System.getenv("GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG"))
-          ? Boolean.parseBoolean(System.getenv("GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG"))
-          : Boolean.parseBoolean(
-              System.getProperty("io.grpc.xds.experimentalEnableCustomLbConfig"));
+      Strings.isNullOrEmpty(System.getenv("GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG"))
```

Comment:
This should be equivalent and seems easier to read:
```java
    Strings.isNullOrEmpty(System.getenv("GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG"))
        || Boolean.parseBoolean(System.getenv("GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG"))
        && Boolean.parseBoolean(System.getProperty(
                "io.grpc.xds.experimentalEnableCustomLbConfig", "true"));
```

But it seems like the getenv part should have parens around it as otherwise the experimentalEnableCustomLbConfig part is basically useless (the only way it could have an effect is if `GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG=true` explicitly).

But it also seems we don't need experimentalEnableCustomLbConfig at all; it looks like you added that because you copied enableLeastRequest. Most of the time we only have environment variables though; the system properties are outliers. So I think we could just have:

```java
    Strings.isNullOrEmpty(System.getenv("GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG"))
        || Boolean.parseBoolean(System.getenv("GRPC_EXPERIMENTAL_XDS_CUSTOM_LB_CONFIG"));
```

Reply:
If we don't actually need to support the system property, then your second suggestion is the way to go. Changed it to use that.